test/integration/related-tasks/profile-touched.js

```diff
@@ -0,0 +1,168 @@
+        ];
+
+        return request(this.workflow)
+          .get(`/related-tasks?model=profile-touched&modelId=${holc.id}&establishmentId=100&limit=10000`)
+          .expect(200)
+          .expect(response => {
+            assertTasks(expected, response.body.data);
+          });
+      });
+
+      it('returns all tasks a user interacted with at an establishment the admin is an admin at', () => {
+        const can = sinon.stub().withArgs('profile.relatedTasks', { id: sinon.match.any, establishment: '100' }).resolves(true);
+        this.workflow.setUser({ profile: holc, can });
+
+        const expected = [
+          'project at Croydon',
+          'granted nio role at croydon'
+          // 'project at Marvell' should not appear
+        ];
+
+        return request(this.workflow)
+          .get(`/related-tasks?model=profile-touched&modelId=${userAtMultipleEstablishments.id}&establishmentId=100&limit=10000`)
+          .expect(200)
+          .expect(response => {
+            assertTasks(expected, response.body.data);
+          });
+      });
+
+      it('throws an unauthorised error for establishments the user is not an admin at', () => {
+        const can = sinon.stub().withArgs('profile.relatedTasks', { id: sinon.match.any, establishment: '101' }).resolves(false);
+        this.workflow.setUser({ profile: holc, can });
+
+        return request(this.workflow)
+          .get(`/related-tasks?model=profile-touched&modelId=${userAtMultipleEstablishments.id}&establishmentId=101&limit=10000`)
+          .expect(403)
+          .expect(response => {
+            assert.equal(response.body.message, `you do not have permission to view related tasks for profile: ${userAtMultipleEstablishments.id}`);
+          });
+      });
+    });
+
+    describe('Asru user', () => {
+      beforeEach(() => {
+        const can = sinon.stub().withArgs('profile.relatedTasks', sinon.match.any).resolves(true);
+        this.workflow.setUser({ profile: licensing, can });
+      });
+
+      it('returns all tasks interacted with by the user at all establishments', () => {
+        const expected = [
+          'pil transfer recalled',
+          'project at Croydon',
+          'granted nio role at croydon',
+          'project at Marvell'
+        ];
+
+        return request(this.workflow)
+          .get(`/related-tasks?model=profile-touched&modelId=${userAtMultipleEstablishments.id}&limit=10000`)
+          .expect(200)
+          .expect(response => {
+            assertTasks(expected, response.body.data);
+          });
+      });
+
+      it('includes tasks that have been actioned by the target profile', () => {
+        const expected = [
+          'discarded by asru'
+        ];
+
+        return request(this.workflow)
+          .get(`/related-tasks?model=profile-touched&modelId=${asruSuper.id}&limit=10000`)
+          .expect(200)
+          .expect(response => {
+            assertTasks(expected, response.body.data);
+          });
+      });
```

Comment:
The test above is the only new test - the rest have just been re-organised cause that test file was mammoth.